include/picongpu/fields/absorber/Absorber.hpp

```diff
@@ -0,0 +1,238 @@
+     *
+     * The general version uses exponential absorber settings since this is the
+     * default absorber.
+     *
+     * @tparam T_FieldSolver field solver
+     */
+    template< typename T_FieldSolver >
+    struct Absorber
+    {
+        //! Number of absorber cells along the min x boundary
+        static constexpr uint32_t xNegativeNumCells = ABSORBER_CELLS[ 0 ][ 0 ];
+
+        //! Number of absorber cells along the max x boundary
+        static constexpr uint32_t xPositiveNumCells = ABSORBER_CELLS[ 0 ][ 1 ];
+
+        //! Number of absorber cells along the min y boundary
+        static constexpr uint32_t yNegativeNumCells = ABSORBER_CELLS[ 1 ][ 0 ];
+
+        //! Number of absorber cells along the max y boundary
+        static constexpr uint32_t yPositiveNumCells = ABSORBER_CELLS[ 1 ][ 1 ];
+
+        //! Number of absorber cells along the min z boundary
+        static constexpr uint32_t zNegativeNumCells = ABSORBER_CELLS[ 2 ][ 0 ];
+
+        //! Number of cells along the max z boundary
+        static constexpr uint32_t zPositiveNumCells = ABSORBER_CELLS[ 2 ][ 1 ];
+
+        //! Get string properties of the absorber
+        static pmacc::traits::StringProperty getStringProperties()
+        {
+            return ExponentialDamping::getStringProperties();
+        }
+    };
+
+    namespace pml = maxwellSolver::yeePML;
+
+    /** Absorber wrapper
+     *
+     * Specialization for PML
+     *
+     * @tparam T_CurrentInterpolation current interpolation for YeePML
+     * @tparam T_CurlE curl E for YeePML
+     * @tparam T_CurlB curl B for YeePML
+     */
+    template<
+        typename T_CurrentInterpolation,
+        typename T_CurlE,
+        typename T_CurlB
+    >
+    struct Absorber<
+        maxwellSolver::YeePML<
+            T_CurrentInterpolation,
+            T_CurlE,
+            T_CurlB
+        >
+    >
+    {
+        //! Number of absorber cells along the min x boundary
+        static constexpr uint32_t xNegativeNumCells = pml::NUM_CELLS[ 0 ][ 0 ];
+
+        //! Number of absorber cells along the max x boundary
+        static constexpr uint32_t xPositiveNumCells = pml::NUM_CELLS[ 0 ][ 1 ];
+
+        //! Number of absorber cells along the min y boundary
+        static constexpr uint32_t yNegativeNumCells = pml::NUM_CELLS[ 1 ][ 0 ];
+
+        //! Number of absorber cells along the max y boundary
+        static constexpr uint32_t yPositiveNumCells = pml::NUM_CELLS[ 1 ][ 1 ];
+
+        //! Number of absorber cells along the min z boundary
+        static constexpr uint32_t zNegativeNumCells = pml::NUM_CELLS[ 2 ][ 0 ];
+
+        //! Number of absorber cells along the max z boundary
+        static constexpr uint32_t zPositiveNumCells = pml::NUM_CELLS[ 2 ][ 1 ];
+
+        //! Get string properties of the absorber
+        static pmacc::traits::StringProperty getStringProperties()
+        {
+            return getStringProperties( "convolutional PML" );
```

Comment:
Note: I should probably clarify here that we can't call `YeePML::getStringProperties()`, since it provides Maxwell's solver properties, as currently the solver and PML are together in that class template.